File: CS-Examples/24_Workbook/MergeExcelFiles.vb
Imports Spire.Xls
Imports System.ComponentModel
Imports System.Text

Namespace MergeExcelFiles
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            ' Create a new List to store file paths
            Dim files As New List(Of String)()

            ' Add the first file path to the list
            files.Add("..\..\..\..\..\..\Data\MergeExcelFiles-1.xlsx")

            ' Add the second file path to the list
            files.Add("..\..\..\..\..\..\Data\MergeExcelFiles-2.xls")

            ' Add the third file path to the list
            files.Add("..\..\..\..\..\..\Data\MergeExcelFiles-3.xlsx")

            ' Create a new Workbook object for the merged Excel files
            Dim workbook As New Workbook()

            ' Set the Excel version of the new workbook to Version2013
            workbook.Version = ExcelVersion.Version2013

            ' Clear any existing worksheets in the new workbook
            workbook.Worksheets.Clear()

            ' Create a temporary Workbook object
            Dim tempbook As New Workbook()

            ' Iterate over each file path in the list
            For Each file As String In files
                ' Load the current file into the temporary workbook
                tempbook.LoadFromFile(file)

                ' Iterate over each worksheet in the temporary workbook
                For Each sheet As Worksheet In tempbook.Worksheets
                    ' Copy each worksheet from the temporary workbook to the new workbook
                    workbook.Worksheets.AddCopy(sheet, WorksheetCopyType.CopyAll)
                Next sheet
            Next file

            ' Save the merged workbook to the specified file path with Excel version compatibility set to Version2010
            workbook.SaveToFile("MergeExcelFiles.xlsx", ExcelVersion.Version2010)

            ' Release the resources used by the workbook
            workbook.Dispose()

            ExcelDocViewer("MergeExcelFiles.xlsx")
		End Sub

		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace

File: CS-Examples/24_Workbook/MergeExcelFiles.cs
using Spire.Xls;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace MergeExcelFiles
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Create the list to add file
            List<string> files = new List<string>();
            files.Add(@"..\..\..\..\..\..\Data\MergeExcelFiles-1.xlsx");
            files.Add(@"..\..\..\..\..\..\Data\MergeExcelFiles-2.xls");
            files.Add(@"..\..\..\..\..\..\Data\MergeExcelFiles-3.xlsx");

            // // Create a workbook with Version2013
            Workbook newbook = new Workbook();
            newbook.Version = ExcelVersion.Version2013;

            //Clear all worksheets
            newbook.Worksheets.Clear();

            // Create a temp workbook
            Workbook tempbook = new Workbook();

            foreach (string file in files)
            {
                // Load the file
                tempbook.LoadFromFile(file);
                foreach (Worksheet sheet in tempbook.Worksheets)
                {
                    //Copy every sheet in a workbook
                    newbook.Worksheets.AddCopy(sheet,WorksheetCopyType.CopyAll);
                }
                // Dispose of the workbook object to release resources
                tempbook.Dispose();
            }

            //Save the file
            newbook.SaveToFile("MergeExcelFiles.xlsx", ExcelVersion.Version2010);

            // Dispose of the workbook object to release resources 
            newbook.Dispose();

            // Launch the file
            ExcelDocViewer("MergeExcelFiles.xlsx");

        }

        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}
